From e6bf065193b231e2d24edd6d239ff4a98d2428b4 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 13 Mar 2019 13:02:19 +0000
Subject: [PATCH 0953/1077] MANTIS 0053707

OpenOrb: NullPointerException in OpenOrb findNextBinding

check if m_bindings != null before accessing m_bindings

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1443 90b65887-3827-0410-9a23-83215b262276
---
 .../OpenORB/src/main/org/openorb/CORBA/Delegate.java        | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/CORBA/Delegate.java b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/CORBA/Delegate.java
index 2327560e..cba8ff03 100644
--- a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/CORBA/Delegate.java
+++ b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/CORBA/Delegate.java
@@ -2270,7 +2270,9 @@ public class Delegate
             // paused channels.
             for (int j = 0, i = getIdx(); j < m_used.length; ++j, i = (i + 1) % m_used.length)
             {
-                if (!m_used[i] && (m_bindings[i].local_invoke()
+                if (!m_used[i]
+                  && m_bindings[i] != null
+                  && (m_bindings[i].local_invoke()
                       || m_bindings[i].getClientChannel().state()
                       == ClientChannel.STATE_CONNECTED))
                 {
@@ -2283,7 +2285,7 @@ public class Delegate
             for (int j = 0, i = getIdx(); j < m_used.length;
                     ++j, i = (i + 1) % m_used.length)
             {
-                if (!m_used[i])
+                if (!m_used[i] && m_bindings[i] != null )
                 {
                     if (m_bindings[i].getClientChannel().state()
                           != ClientChannel.STATE_CLOSED)
-- 
2.41.0.windows.1

